src/test/java/seedu/recruittrackpro/logic/commands/EditTagCommandTest.java

```diff
@@ -0,0 +1,79 @@
+package seedu.recruittrackpro.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.recruittrackpro.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.recruittrackpro.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.recruittrackpro.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.recruittrackpro.testutil.TypicalPersons.getTypicalRecruitTrackPro;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.recruittrackpro.commons.core.index.Index;
+import seedu.recruittrackpro.model.Model;
+import seedu.recruittrackpro.model.ModelManager;
+import seedu.recruittrackpro.model.RecruitTrackPro;
+import seedu.recruittrackpro.model.UserPrefs;
+import seedu.recruittrackpro.model.person.Person;
+import seedu.recruittrackpro.model.tag.Tag;
+import seedu.recruittrackpro.testutil.PersonBuilder;
+
+public class EditTagCommandTest {
+    private Model model = new ModelManager(getTypicalRecruitTrackPro(), new UserPrefs());
+
+    @Test
+    public void execute_validTagEdit_success() {
+        Person targetPerson = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person updatedPerson = new PersonBuilder(targetPerson)
+                .removeTags("friends").addTags("besties").build();
```

Comment:
Referencing earlier's suggestion, we can create a new `editTag()` method in `PersonBuilder` and encapsulate this 

```suggestion
        Person updatedPerson = new PersonBuilder(targetPerson)
                .editTag("friends", "besties").build();
```

In `PersonBuilder`, add the following:
```suggestion
/**
 * Replaces an existing tag with a new tag for the {@code Person} being built.
 */
public PersonBuilder replaceTag(String oldTag, String newTag) {
    try {
        Tags updatedTags = this.tags.replaceTag(new Tag(oldTag), new Tag(newTag));
        this.tags = updatedTags;
    } catch (ParseException e) {
        throw new IllegalArgumentException("Tag replacement failed: " + e.getMessage(), e);
    }
    return this;
}